### PR TITLE
store: lock stores before calling Exists

### PR DESCRIPTION
**Pull request by giuseppe**
we need to lock the stores for read before checking if a layer exists.

follow up for 466ee04ed72429867df1fea80d992722ab571415

addressing the comment here: https://github.com/containers/storage/pull/1873#discussion_r1553620140

### CODE REVIEW

File: store.go [hunk 1]
```diff
@@ -2824,21 +2824,28 @@ func (s *store) mount(id string, options drivers.MountOpts) (string, error) {
...
 
-	if rlstore.Exists(id) {
+	if err := rlstore.startWriting(); err != nil {
+		return "", err
+	}
+	exists := rlstore.Exists(id)
+	rlstore.stopWriting()
+
+	if exists {
 		return rlstore.Mount(id, options)
```
Comment (mtrmac) [inline store.go:2838]: This also requires a write lock.

**Review by mtrmac — COMMENTED** [review]
Thanks for the followup!

File: store.go (hunk 1 above)
Comment (giuseppe) [reply store.go]: ouch, right.

Fixed now

**Review by mtrmac — APPROVED** [review]
/lgtm

**Pull request merged**